src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -246,22 +231,22 @@ public async Task<ImmutableArray<DiagnosticData>> GetProjectDiagnosticsAsync(Can
                 if (ProjectId != null)
                 {
                     var project = Solution.GetProject(ProjectId);
-                    if (project != null)
-                    {
-                        await AppendDiagnosticsAsync(project, documentIds: [], includeProjectNonLocalResult: true, cancellationToken).ConfigureAwait(false);
-                    }
+                    if (project is null)
+                        return [];
 
-                    return GetDiagnosticData();
+                    return await ProduceProjectDiagnosticsAsync(
+                        [project], static _ => [], includeProjectNonLocalResult: true, cancellationToken).ConfigureAwait(false);
```

Comment:
note: thsi logic intentionally changed.  the old logic said "if you give me a project id not in the solution, i give you all solution diagnostics". t he new logic says "if you give me a project not in the solution, you get back nothing".  this mirrors the logic higher up in this file.